code/game/machinery/vending.dm

```diff
@@ -2378,3 +2378,46 @@ var/global/num_vending_terminals = 1
  		/obj/item/clothing/suit/armor/knight/templar = 5,
 		)
 	pack = /obj/structure/vendomatpack/chapelvend
+
+
+/obj/machinery/vending/trader	// Boxes are defined in trader.dm
+	name = "Jewvend"
+	desc = "Its coin groove has been modified."
+	product_slogans = "Profits."
+	product_ads = "When you charge a customer $100, and he pays you by mistake $200, you have an ethical dilemma — should you tell your partner?"
+	vend_reply = "Money money money!"
+	icon_state = "voxseed"
+	products = list (
+		/obj/item/weapon/storage/fancy/donut_box = 2,
+		)
+
+/obj/machinery/vending/trader/New()
+	..()
+	premium = list(
+		/obj/item/weapon/storage/trader_marauder,
+		/obj/item/weapon/storage/backpack/holding,
+		/obj/item/weapon/reagent_containers/glass/beaker/bluespace,
+		/obj/item/weapon/storage/bluespace_crystal,
+		/obj/item/clothing/shoes/magboots/elite,
+		/obj/item/weapon/reagent_containers/food/snacks/borer_egg,
+		/obj/item/weapon/reagent_containers/glass/bottle/peridaxonsmall,
+		/obj/item/weapon/reagent_containers/glass/bottle/rezadone,
+		/obj/item/weapon/reagent_containers/glass/bottle/nanitessmall,	
+		)
+
+	for(var/random_items = 1 to 5)
```

Comment:
If you add more items to the premium list you'd have to modify the 5 here, use `premium.len - #` or add to the premium list instead